libraries/chain/combined_database.cpp

```diff
@@ -308,9 +296,9 @@ namespace eosio { namespace chain {
          db.commit(revision);
          break;
       case backing_store_type::ROCKSDB:
+         db.commit(revision);
          try {
             try {
-               db.commit(revision);
                kv_undo_stack->commit(revision);
```

Comment:
This approach is fine, but I think removing the break and not duplicating the logic would be clearer. Here and all the other switch statements.

Reply:
I thought about refactoring that code too, but was afraid of making too big changes to impact existing outstanding branches. Will make the changes.